File: tests/machine/x/pl/string_prefix_slice.pl
:- style_check(-singleton).
slice(Str, I, J, Out) :-
    string(Str), !,
    Len is J - I,
    sub_string(Str, I, Len, _, Out).
slice(List, I, J, Out) :-
    length(Prefix, I),
    append(Prefix, Rest, List),
    Len is J - I,
    length(Out, Len),
    append(Out, _, Rest).

len_any(Value, Len) :-
    string(Value), !, string_length(Value, Len).
len_any(Value, Len) :-
    is_dict(Value), !, dict_pairs(Value, _, Pairs), length(Pairs, Len).
len_any(Value, Len) :- length(Value, Len).

:- initialization(main, main).
main :-
    Prefix = "fore",
    S1 = "forest",
    len_any(Prefix, _V0),
    slice(S1, 0, _V0, _V1),
    ((_V1 == Prefix) -> _V2 = true ; _V2 = false),
    writeln(_V2),
    S2 = "desert",
    len_any(Prefix, _V3),
    slice(S2, 0, _V3, _V4),
    ((_V4 == Prefix) -> _V5 = true ; _V5 = false),
    writeln(_V5),
    true.

File: tests/aster/x/lua/string_prefix_slice.lua
prefix = "fore";
s1 = "forest";
print((string.sub(s1, (0 + 1), (function(v)
  if type(v) == 'table' and v.items ~= nil then
    return #v.items;
  elseif type(v) == 'table' and (v[1] == nil) then
    local c = 0;
    for _ in pairs(v) do
      c = c + 1;
    end;
    return c;
  else
    return #v;
  end;
end)(prefix)) == prefix));
s2 = "desert";
print((string.sub(s2, (0 + 1), (function(v)
  if type(v) == 'table' and v.items ~= nil then
    return #v.items;
  elseif type(v) == 'table' and (v[1] == nil) then
    local c = 0;
    for _ in pairs(v) do
      c = c + 1;
    end;
    return c;
  else
    return #v;
  end;
end)(prefix)) == prefix));
